skbio/stats/composition.py

```diff
@@ -608,10 +611,299 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0-dev")
+def ancom(table, grouping,
+          alpha=0.05,
+          tau=0.02,
+          theta=0.1,
+          multiple_comparisons_correction=None,
+          significance_test=None):
+    r""" Performs a differential abundance test using ANCOM
+
+    This is done by calculating pairwise log ratios between all features
+    and performing a signficance test to determine if there is a significant
+    difference in feature ratios with respect to the variable of interest.
+
+    In an experiment with only two treatments, this test tests the following
+    hypothesis for feature :math:`i`
+
+    .. math::
+
+        H_{0i}: \mathbb{E}[\ln(u_i^{(1)})] = \mathbb{E}[\ln(u_i^{(2)})]
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature :math:`i` in the
+    first group and :math:`u_i^{(2)}` is the mean abundance for feature
+    :math:`i` in the second group.
+
+    This method can be extended to an arbitrary number of classes by passing
+    in a different statistical function.
+
+    Parameters
+    ----------
+    table : pd.DataFrame
+       A 2D matrix of counts or proportions where the rows correspond to
+       samples and the columns correspond to features
+    grouping : pd.Series
+       Vector of group categories
+    alpha : float, optional
+       Significance level for each of the statistical tests
+    tau : float, optional
+       A constant used to determine an appropriate cutoff
+    theta : float, optional
+       Lower bound for the proportion of differential features.
+       This can can be anywhere between 0 and 1.
+    multiple_comparisons_correction : {None, 'holm-bonferroni'}, optional
+       The multiple comparison correction procedure to run.  By default
+       `scipy.stats.f_oneway` is used.
+    significance_test : function, optional
+       A statistical signficance function to test for signficance between
+       classes.
+
+    Returns
+    -------
+    pd.DataFrame
+        A table of features, their W-statistics and whether the null hypothesis
+        is rejected.
+
+        `"W"` is the W-statistic, or number of features that a single feature
+        is tested to be significantly different against.
+
+        `"reject"` indicates if feature is significantly different or not
+
+    See Also
+    --------
+    multiplicative_replacement
+
+    Notes
+    -----
+    This method cannot handle any zero counts as input, since the logarithm
+    of zero cannot be computed.  While this is an unsolved problem, many
+    studies have shown promising results by replacing the zeros with pseudo
+    counts.
```

Comment:
Can you also describe `multiplicative_replacement` here since that is also a viable option?